### PR TITLE
Add basic MVCC tests

### PR DESCRIPTION
**Pull request by krishnangovindraj**
## Usage and product changes
Adds basic tests ensuring we can open views into past versions of the database & that commits increment the (default) version seen by new transactions.

## Implementation
* Adds test cases to `test_mvcc.rs`

### CODE REVIEW

**Review by krishnangovindraj — COMMENTED** [review]
Self review comments.

File: storage/tests/test_mvcc.rs [hunk 1]
```diff
@@ -34,3 +34,137 @@ This file should comprise a set of low-level tests relating to MVCC.
...
+    snapshot_read_01.close_resources();
+    snapshot_read_0.close_resources();
+
+    // Read from further in the past.
+    let snapshot_read_02 = storage.open_snapshot_read_at(watermark_0);
+    assert_eq!(snapshot_read_02.get::<128>(key_1.as_reference()).unwrap().unwrap().bytes(), VALUE_0);
+    snapshot_read_02.close_resources();
+}
+
+#[test]
```
Comment (krishnangovindraj) [inline storage/tests/test_mvcc.rs:137]: I'm not sure this test is meaningful in our context given

1. We may not need to fail transactions which concurrently write the same key.
2. We have isolation tests to test isolation is enforced.

File: storage/tests/test_mvcc.rs [hunk 2]
```diff
@@ -34,3 +34,137 @@ This file should comprise a set of low-level tests relating to MVCC.
...
+fn test_commit_increments_watermark() {
+    init_logging();
+    let storage_path = create_tmp_dir();
+    let storage = setup_storage(&storage_path);
+    let wm_initial = storage.read_watermark();
+    let snapshot_0 = storage.open_snapshot_write();
+    snapshot_0.commit().unwrap();
+
+    assert_eq!(wm_initial.next().number().number() + 1, storage.read_watermark().number().number());
```
Comment (krishnangovindraj) [inline storage/tests/test_mvcc.rs:94]: Currently fails.

**Review by krishnangovindraj — COMMENTED** [review]
New self review after moving everything back to test_mvcc.rs

File: storage/tests/test_mvcc.rs [hunk 3]
```diff
@@ -34,3 +34,137 @@ This file should comprise a set of low-level tests relating to MVCC.
    After cleanup is run, if we iterate directly on the storage layer, we should be able to confirm the keys are actually not present anymore (Rocks may defer the disk delete till compaction, but to us they are "gone").
 
  */
+
```
Comment (krishnangovindraj) [inline storage/tests/test_mvcc.rs:37]: Comment #4 above is still pending.
If we're happy with the current case, I'll move it to a TODO comment.

File: storage/tests/test_mvcc.rs (hunk 1 above)
Comment (flyingsilverfin) [reply storage/tests/test_mvcc.rs:137]: Hang on, just from reading the title, in our semantics writing the same key twice should not fail right?

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**